features/cellular/framework/device/CellularStateMachine.h

```diff
@@ -161,7 +165,7 @@ class CellularStateMachine {
 
     Callback<void(nsapi_event_t, intptr_t)> _event_status_cb;
 
-    CellularNetwork *_network;
+    CellularNetwork &_network;
```

Comment:
how is this related to adding new connection status ? separate commit with explanation?

Reply:
removed, will do this fix in a different pr.